game/src/core/hud/HeroUI.java

```diff
@@ -0,0 +1,181 @@
+package core.hud;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.g2d.BitmapFont;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.graphics.g2d.freetype.FreeTypeFontGenerator;
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.actions.Actions;
+
+import contrib.components.HealthComponent;
+import contrib.components.XPComponent;
+
+import core.Game;
+import core.utils.Constants;
+import core.utils.Point;
+import core.utils.controller.ScreenController;
+
+/** This class represents the UI of the hero */
+public class HeroUI<T extends Actor> extends ScreenController<T> {
+
+    private static final HeroUI<Actor> heroUI = new HeroUI<>(new SpriteBatch());
+    private ScreenText level;
+    private ScreenImage healthBar, xpBar;
+    private BitmapFont font;
+
+    private record HeroData(HealthComponent hc, XPComponent xc) {}
+
+    /**
+     * Creates a Screencontroller with a ScalingViewport which stretches the ScreenElements on
+     * resize
+     *
+     * @param batch the batch which should be used to draw with
+     */
+    private HeroUI(SpriteBatch batch) {
+        super(batch);
+        setup();
+    }
+
+    /** Updates the UI with the current data of the hero */
+    public void updateUI(HeroData hd) {
```

Comment:
Bitte in `update` umbenennen, damit es einheitlich ist (wir verwenden überall den Namen update) 
Und wie oben schon gesagt, würde ich mir `HeroData hd` einfach in der Methode zusammenbauen und keinen Parameter haben. 
Und mein Bauchgefühl sagt mir, hier könnte man dann auch den Code zum überprüfen der Components und ggf. das triggern der HUD-Elemente implementieren.

Reply:
Ich habs mal geändert, die popup calls sind jetzt komplett raus aus HealthComponent und HealthSystem und werden jetzt mit in HeroUI gemacht.  Musst du mal gucken ob es jetzt besser ist als vorher.

Reply:
macht auf mich einen guten Eindruck, danke ^^